core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/WhineyConsumer.java

```diff
@@ -3,16 +3,14 @@
 import java.io.IOException;
 import java.util.function.Consumer;
 
-import org.slf4j.Logger;
-
 public interface WhineyConsumer<T> {
 
-  public static <T> Consumer<T> log(Logger logger, WhineyConsumer<? super T> function) {
+  public static <T> Consumer<T> rethrow(WhineyConsumer<? super T> function) {
     return arg -> {
       try {
         function.accept(arg);
       } catch (IOException e) {
-        logger.error("Ignoring error", e);
+        throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)